app/controllers/order_details_controller.rb

```diff
@@ -72,6 +72,27 @@ def dispute
 
   private
 
+  def cancellation_success_path(order_detail)
+    if operator_cancellation?(order_detail)
+      facility_user_reservations_path(order_detail.facility, order_detail.user)
+    else
+      reservations_path
+    end
+  end
+
+  def cancel_with_fee(order_detail)
+    if operator_cancellation?(order_detail)
+      # When this is an operator who is not the ordering user; admin cancel with fee:
+      order_detail.cancel_reservation(session_user, OrderStatus.canceled_status, true, true)
+    else
+      order_detail.cancel_reservation(session_user)
+    end
+  end
+
+  def operator_cancellation?(order_detail)
+    order_detail.user != session_user && current_ability.can?(:cancel, order_detail)
```

Comment:
This is probably an edge case, but if I'm an operator who has placed an order for myself, I think it will always bring me back to the operator view, even if I cancel from the user view.